client/src/main/java/org/apache/uniffle/client/impl/ShuffleWriteClientImpl.java

```diff
@@ -192,12 +206,27 @@ private boolean sendShuffleDataAsync(
     return result;
   }
 
-  private void genServerToBlocks(ShuffleBlockInfo sbi, List<ShuffleServerInfo> serverList,
-      Map<ShuffleServerInfo, Map<Integer, Map<Integer, List<ShuffleBlockInfo>>>> serverToBlocks,
-      Map<ShuffleServerInfo, List<Long>> serverToBlockIds) {
+  void genServerToBlocks(ShuffleBlockInfo sbi,
+                         List<ShuffleServerInfo> serverList,
+                         int replicaNum,
+                         List<ShuffleServerInfo> excludeServers,
+                         Map<ShuffleServerInfo, Map<Integer, Map<Integer, List<ShuffleBlockInfo>>>> serverToBlocks,
+                         Map<ShuffleServerInfo, List<Long>> serverToBlockIds,
+                         boolean includeBlockList) {
```

Comment:
Why is it always negated in line 223, should we name it `excludeBlocklist` instead?
i.e. change `if (!includeBlockList && ...` to `if (excludeBlocklist && ...`

Reply:
Speaking of the naming, I think `with/without` sounds better than `include/exclude`.
For example, `withBlocklist`, `withoutBlocklist`.

Reply:
Done